### ISSUE TITLE
COLLATION placeholder not available in upgrade SQL scripts

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#20102]
#### Describe the bug

PrestaShop relies on placeholders for its SQL install and upgrade scripts (see https://devdocs.prestashop.com/1.7/modules/creation/enabling-auto-update/).

This means that for example query
```
CREATE TABLE `PREFIX_customer_session` (
  `id_customer_session` int(11) unsigned NOT NULL auto_increment,
  `id_customer` int(10) unsigned DEFAULT NULL,
  `token` varchar(40) DEFAULT NULL,
  PRIMARY KEY `id_customer_session` (`id_customer_session`)
) ENGINE=ENGINE_TYPE DEFAULT CHARSET=utf8;
```
is, at runtime, transformed as
```
CREATE TABLE `PREFIX_customer_session` (
  `id_customer_session` int(11) unsigned NOT NULL auto_increment,
  `id_customer` int(10) unsigned DEFAULT NULL,
  `token` varchar(40) DEFAULT NULL,
  PRIMARY KEY `id_customer_session` (`id_customer_session`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

In install SQL scripts, 3 placeholders are available:
- ENGINE_TYPE
- PREFIX_
- COLLATION

But in SQL upgrade scripts, the COLLATION placeholder is not available. We learnt it [the hard way when testing 1.7.6.6 release](https://github.com/PrestaShop/PrestaShop/issues/20013).

We should enable the use of COLLATION placeholder in upgrade scripts too.

However, this is tricky. Because if we allow this placeholder to work in 1.7.7.0, for example, then _shops upgrading from 1.7.7.0 to 1.7.7.1_ will be able to use it but _not shops using 1.7.6.5 to 1.7.7.1_, because this change will be in 1.7.7.0 Core.

So we have to think hard about compatibility range here 😅

**Comment by marionf** [comment]
@matks I don't how important this is, should we fix it for 1.7.7.0 ?

**Comment by matks** [comment]
> @matks I don't how important this is, should we fix it for 1.7.7.0 ?

No, it's not critical. It would be a nice improvement for prestashop in the long term.
